veps/sig-compute/target-xml-mutation-hooks.md

```diff
@@ -0,0 +1,120 @@
+# VEP 141: Target-Side Pre-Migration Hooks for Domain XML Modification
+
+## Release Signoff Checklist
+
+- [X] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+- [ ] (R) Target version is explicitly mentioned and approved
+- [ ] (R) Graduation criteria filled
+
+## Overview
+
+KubeVirt currently applies domain XML transformations on the source launcher pod during live migration.  
+This proposal introduces a target-side pre-migration hook system that performs all mutations on the target launcher pod before the migration start.
+
+## Motivation
+
+Currently, KubeVirt modifies the domain XML on the source during live migration via `migratableDomXML()`. This approach has limitations:
+
+1. **Upgrade**  
+   - New XML transformation logic requires two migrations: old source → new target, then new source → new target.  
+   - Users might not even know a second migration is required, and no automation exists to trigger it during upgrades.
+
+2. **Information Gathering Constraints**  
+   - CPUSet and topology information are currently added to the VMI status, filling our API with implementation details.  
+   - This approach does not scale as more target-specific information is added through the vmi.status API.
+
+3. **Blocked VEPs**  
+   Some VEPs also require changes to the domain XML and aim to avoid source-based modifications. This hook system can  
+   serve as a base that they can integrate with to modify XML in a simple and composable way.  
+   Example of such VEPs:  
+   - vGPU live migration [(VEP 109)](https://github.com/kubevirt/enhancements/issues/110)  
+   - Secondary network upgrades [(VEP 111)](https://github.com/kubevirt/enhancements/issues/111)
+
+## Goals
+
+- Move all XML mutation logic to the target pod
+- Introduce a lightweight hook system owned by SIGs
```

Comment:
This sounds more like an implementation detail than a high-level goal. How about discussing it in the design section instead?